platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronisedInstallerImpl.java

```diff
@@ -0,0 +1,566 @@
+      }
+    } catch (IOException e) {
+      throw new SynchronisedInstallerException(
+          "Failed to update created managed service factory configuration with pid of ["
+              + createdConfig.getPid()
+              + "]",
+          e);
+    }
+
+    waitForServiceAvailability(maxWaitTime, createdConfig.getPid());
+    return createdConfig;
+  }
+
+  @Override
+  public void updateManagedService(
+      String servicePid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    updateManagedService(DEFAULT_MAX_SERVICE_WAIT, servicePid, properties, bundleLocation);
+  }
+
+  @Override
+  public void updateManagedService(
+      long maxWaitTime, String servicePid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    waitForServiceAvailability(maxWaitTime, servicePid);
+
+    LOGGER.debug(
+        "Registering service configuration listener to listen to service with pid of {}.",
+        servicePid);
+    ServiceConfigurationListener listener = new ServiceConfigurationListener(servicePid);
+    ServiceRegistration<?> registration =
+        bundleContext.registerService(ConfigurationListener.class.getName(), listener, null);
+
+    Callable<Boolean> isServiceUpdated =
+        () -> {
+          LOGGER.debug(
+              "Waiting for service with pid [{}] to reflect configuration updates...", servicePid);
+          return listener.isUpdated();
+        };
+
+    boolean available;
+
+    LOGGER.info("Updating configuration of service with pid [{}].", servicePid);
+
+    try {
+      Configuration config = configAdmin.getConfiguration(servicePid, bundleLocation);
+      config.setBundleLocation(bundleLocation);
+      config.update(toDic(properties));
+
+      LOGGER.debug("Updated configuration of service with pid [{}].", servicePid);
+      available = wait(isServiceUpdated, maxWaitTime, DEFAULT_POLLING_INTERVAL);
+    } catch (IOException e) {
+      throw new SynchronisedInstallerException(
+          "Failed to update managed service configuration with pid of [" + servicePid + "]", e);
+    } catch (Exception e) {
+      throw new SynchronisedInstallerException(
+          "Interrupted while waiting for service with pid of [" + servicePid + "] to be updated.",
+          e);
+    } finally {
+      registration.unregister();
+    }
+
+    if (!available) {
+      throw new SynchronisedInstallerException(
+          "Managed service failed to update within [" + maxWaitTime + "] ms.");
+    }
+  }
+
+  @Override
+  public void waitForServiceAvailability(String servicePid) throws SynchronisedInstallerException {
+    waitForServiceAvailability(DEFAULT_MAX_SERVICE_WAIT, servicePid);
+  }
+
+  @Override
+  public void waitForServiceAvailability(long maxWaitTime, String servicePid)
+      throws SynchronisedInstallerException {
+    ServiceTracker st = getServiceTracker(bundleContext, getServiceIdFilter(servicePid));
```

Comment:
❗️ This assumes that the bundle in which this class is defined has permissions to monitor any or all services. the policy file should be updated and privileges should be extended for this to work.